### PR TITLE
feat: add Cohere Command model provider

### LINKED ISSUE #696: feat: support Cohere "command" model
https://docs.cohere.com/docs/models

Command model is accessible via the cohere Chat API https://docs.cohere.com/reference/chat

### PR DESCRIPTION
**Pull request by MartinRepo**
Fix: https://github.com/casibase/casibase/issues/696

**For supporting Cohere "command" model, I modified 5 files as follows:**
1. Added casibase/model/cohere.go, for initializing Command model and use it.
2. Modified casibase/model/provider.go, add condition to check Command model.
3. Modified casibase/model/util.go, add a "User" in RawMessage for the parameter "chat_history" in Command.
4. Modified casibase/web/src/Setting.js, add options for Cohere in Model.
5. Modified casibase/web/src/ProviderEditPages.js, add a default model option for Cohere.

### CODE REVIEW

File: model/util.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 type RawMessage struct {
 	Text   string
 	Author string
+	User   *string
```
Comment (hsluoyz) [inline model/util.go]: Why add this?

File: web/src/Setting.js [hunk 2]
```diff
@@ -648,6 +648,7 @@ export function getProviderTypeOptions(category) {
         {id: "MiniMax", name: "MiniMax"},
         {id: "Local", name: "Local"},
         {id: "Azure", name: "Azure"},
+        {id: "Cohere", name: "Cohere"},
```
Comment (hsluoyz) [inline web/src/Setting.js:651]: Code not added for `category === "Embedding"`?

File: web/src/ProviderEditPage.js [hunk 3]
```diff
@@ -191,6 +191,8 @@ class ProviderEditPage extends React.Component {
                   this.updateProviderField("subType", "custom-model");
                 } else if (value === "Azure") {
                   this.updateProviderField("subType", "gpt-4");
+                } else if (value === "Cohere") {
+                  this.updateProviderField("subType", "command");
```
Comment (hsluoyz) [inline web/src/ProviderEditPage.js:195]: Code not added for `this.state.provider.category === "Embedding"`?

File: model/cohere.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	cohere "github.com/henomis/lingoose/llm/cohere"
+)
+
+type CohereModelProvider struct {
+	secretKey   string
```
Comment (hsluoyz) [inline model/cohere.go]: CohereEmbeddingProvider not added?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: model/cohere.go (hunk 4 above)
Comment (MartinRepo) [reply model/cohere.go]: Cohere Embedding Provider has existed in embedding/cohere.go

File: web/src/ProviderEditPage.js (hunk 3 above)
Comment (MartinRepo) [reply web/src/ProviderEditPage.js:195]: Embedding for Cohere has existed in "Embedding" condition branch

File: web/src/Setting.js (hunk 2 above)
Comment (MartinRepo) [reply web/src/Setting.js:651]: "Embedding" condition branch has existed.

File: model/util.go (hunk 1 above)
Comment (MartinRepo) [reply model/util.go]: I have deleted it.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**